dev/TEAL.md

```diff
@@ -443,57 +548,67 @@ of a true top-level transaction, programatically.  However, they are
+In v5, only a few of the Header fields may be set: `Type`/`TypeEnum`,
+`Sender`, and `Fee`. In v6, Header fields `Note` and `RekeyTo` may
+also be set.  For the specific (non-header) fields of each transaction
+type, any field may be set.  This allows, for example, clawback
+transactions, asset opt-ins, and asset creates in addition to the more
+common uses of `axfer` and `acfg`.  All fields default to the zero
+value, except those described under `itxn_begin`.
 
 Fields may be set multiple times, but may not be read. The most recent
-setting is used when `itxn_submit` executes. (For this purpose `Type`
-and `TypeEnum` are considered to be the same field.) `itxn_field`
-fails immediately for unsupported fields, unsupported transaction
-types, or improperly typed values for a particular field. `itxn_field`
-makes aceptance decisions entirely from the field and value provided,
-never considering previously set fields. Illegal interactions between
-fields, such as setting fields that belong to two different
-transaction types, are rejected by `itxn_submit`.
-
-| Op | Description |
-| --- | --- |
-| `itxn_begin` | begin preparation of a new inner transaction |
-| `itxn_field f` | set field F of the current inner transaction to X |
-| `itxn_submit` | execute the current inner transaction. Fail if 16 inner transactions have already been executed, or if the transaction itself fails. |
-| `itxn f` | push field F of the last inner transaction to stack |
-| `itxna f i` | push Ith value of the array field F of the last inner transaction to stack |
+setting is used when `itxn_submit` executes. For this purpose `Type`
+and `TypeEnum` are considered to be the same field. When using
+`itxn_field` to set an array field (`ApplicationArgs` `Accounts`,
+`Assets`, or `Applications`) each use adds an element to the end of
+the the array, rather than setting the entire array at once.
+
+`itxn_field` fails immediately for unsupported fields, unsupported
+transaction types, or improperly typed values for a particular
+field. `itxn_field` makes aceptance decisions entirely from the field
+and value provided, never considering previously set fields. Illegal
+interactions between fields, such as setting fields that belong to two
+different transaction types, are rejected by `itxn_submit`.
+
+| Opcode | Description |
+| - | -- |
+| `itxn_begin` | begin preparation of a new inner transaction in a new transaction group |
+| `itxn_next` | begin preparation of a new inner transaction in the same transaction group |
+| `itxn_field f` | set field F of the current inner transaction to A |
+| `itxn_submit` | execute the current inner transaction group. Fail if executing this group would exceed the inner transaction limit, or if any transaction in the group fails. |
+| `itxn f` | field F of the last inner transaction |
+| `itxna f i` | Ith value of the array field F of the last inner transaction |
+| `gitxn t f` | field F of the Tth transaction in the last inner group submitted |
```

Comment:
Yes. `itxn` was introduced when there could only be one inner at a time. We keep it around because deprecation is annoying, but it's not really needed anymore.